### ISSUE TITLE
Maskfile Support

### ISSUE THREAD

**Issue by vanjobanjo** [hashview/hashview#87]
The hashview ui supports creating single tasks. Is there a way to upload mask files like https://github.com/golem445/Corporate_Masks provides?

**Comment by i128** [comment]
I like this suggestion. We could swap out the single line entry for either a multi line entry for copy/paste and/or file upload. Similar to how we do hashfiles currently

**Comment by vanjobanjo** [comment]
Yes, that would be great! Given the size of some of these mask files, I would prefer the file upload option. But having both would be perfect.

**Comment by bandrel** [comment]
Still wanted, four years on — and worth linking up, because this request has since fragmented into three other issues that don't reference each other or this one:

- **#87** (this one) — UI: upload a maskfile instead of typing one mask per task.
- **#381** — `POST /v1/tasks/add` is attack-mode-0 only, so a mask task can't be created over the API at all.
- **#384** — there is no `control/masks` store, so a maskfile has nowhere to live server-side and no agent sync path.
- **#401** — no `/v1` endpoints for task groups, so even once the tasks exist, bundling them is UI-only.

Those four are really one feature seen from four angles: get a maskfile onto the server, let a task point at it, let the API create that task, and let the resulting tasks be grouped. #384 is the piece the other three lean on.

## A workaround that works today, if it helps scope the real fix

On a stock 0.8.3 you can get the full Corporate_Masks set (all 7 files, 109,695 masks) onto a server with no code changes, by parking the `.hcmask` files in the **rules** catalog:

1. `POST /v1/rules/add/<name>` with the `.hcmask` as the body. Rules are stored **plaintext** at rest, unlike wordlists, which are gzipped — and gzip is exactly what hashcat can't read in the mask-file argv position. (Send the body gzipped even so; the plaintext path trips the `os.rename` EXDEV bug in #395 when `control/` is a mount.)
2. The agent's `sync_rules()` gzip-decompresses each one into `control/rules/<file>`, so the file lands on every agent as plaintext with no new sync code.
3. Create a Brute-force task whose Mask field is that agent-side path, e.g. `control/rules/86209f7dad67e867.txt`. hashcat takes a mask file as a literal argv path in `-a 3`, so one task runs every mask in the file — 22,859 masks in a single task rather than 22,859 tasks.

Two rough edges that I think are informative for the real implementation rather than just being workaround problems:

- **The server renames uploads to a random `<hex>.txt`**, so the path you need in the Mask field isn't predictable from the name you uploaded — you have to read it back out of `GET /v1/rules`. A first-class maskfile feature probably wants to surface the agent-side path in the UI, since the user otherwise can't know it.
- **`Tasks.hc_mask` is `String(50)`.** `control/masks/<something>` fits, but not by a lot, and it's a hard ceiling on any path-based design.

I have the 7 tasks and a task group built this way on a 0.8.3 instance and verified the paths persist correctly end to end (mode 3, full path intact, no truncation). I have **not** run a job through them yet, so treat "hashcat consumes it" as reasoned from the argv semantics and the plaintext-on-disk guarantee, not as a benchmarked result.

## Offer

If there's interest, I'd rather help build the real thing than keep working around it. The client-side half already exists — I've got tooling that does the upload and emits the paste-ready mask paths — so the missing pieces are server-side and mostly small: the `control/masks` store from #384, `hc_attackmode` acceptance in #381, and a file-upload widget here. Happy to send PRs if a maintainer indicates which shape they'd take (in particular, whether maskfiles get their own store as in #384 or reuse the rules table).
